### PR TITLE
[Fix] Snowflake connection bug of Apache Superset

### LINKED ISSUE #9083: Error while Connecting to Snowflakes
I'm trying to connect to snowflakes. After entering SQLAlchemy URI, I get  error

### Expected results

Test connection should be successful

### Actual results

It throws an error
`ERROR: {"error": "Connection failed!\n\nThe error message returned was:\nCan't load plugin: sqlalchemy.dialects:snowflake", "stacktrace": "Traceback (most recent call last):\n  File \"/app/superset/views/core.py\", line 1358, in testconn\n    engine = database.get_sqla_engine(user_name=username)\n  File \"/app/superset/utils/core.py\", line 131, in __call__\n    value = self.func(*args, **kwargs)\n  File \"/app/superset/models/core.py\", line 318, in get_sqla_engine\n    return create_engine(sqlalchemy_url, **params)\n  File \"/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/__init__.py\", line 479, in create_engine\n    return strategy.create(*args, **kwargs)\n  File \"/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/strategies.py\", line 61, in create\n    entrypoint = u._get_entrypoint()\n  File \"/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/url.py\", line 172, in _get_entrypoint\n    cls = registry.load(name)\n  File \"/usr/local/lib/python3.6/site-packages/sqlalchemy/util/langhelpers.py\", line 240, in load\n    \"Can't load plugin: %s:%s\" % (self.group, name)\nsqlalchemy.exc.NoSuchModuleError: Can't load plugin: sqlalchemy.dialects:snowflake\n"}
`
#### How to reproduce the bug

1. Use docker image to run superset
2. Login -> Sources -> Databases -> Add Database enter URI in SQLAlchemy URI 

### Environment

(please complete the following information):

- superset version: `0.999dev`
### Checklist

Make sure these boxes are checked before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### PR DESCRIPTION
**Pull request by saurabhperiwal**
Fix #9083

### CATEGORY

Choose one

- [x] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
Snowflakes connector was not working because snowflake-sqlalchemy was missing

### ADDITIONAL INFORMATION

Fixes #9083 
- [ ] Has associated issue: #9083 
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -104,6 +104,7 @@ def get_git_sha():
         "simplejson>=3.15.0",
         "sqlalchemy>=1.3.5, <2.0",
         "sqlalchemy-utils>=0.33.2",
+	"snowflake-sqlalchemy>=1.2.0",
```
Comment (villebro) [inline setup.py]: Please fix the indentation here and move to `extras_require`; `snowflake-sqlachemy` is not required to run Superset.

**Review by dpgaspar — CHANGES_REQUESTED** [review]
snowflake is an optional dependency, take a look at: https://superset.incubator.apache.org/installation.html#database-dependencies

File: requirements.txt [hunk 2]
```diff
@@ -60,26 +73,35 @@ prison==0.1.2             # via flask-appbuilder
...
 pyyaml==5.3
+requests==2.22.0          # via azure-storage-common, snowflake-connector-python
 retry==0.9.2
+s3transfer==0.3.2         # via boto3
 selenium==3.141.0
 simplejson==3.17.0
-six==1.14.0               # via bleach, cryptography, flask-jwt-extended, flask-talisman, isodate, jsonschema, pathlib2, polyline, prison, pyarrow, pyrsistent, python-dateutil, sqlalchemy-utils, wtforms-json
+six==1.14.0               # via bleach, cryptography, flask-jwt-extended, flask-talisman, isodate, jsonschema, pathlib2, polyline, prison, pyarrow, pyopenssl, pyrsistent, python-dateutil, sqlalchemy-utils, wtforms-json
+snowflake-connector-python==2.2.0  # via snowflake-sqlalchemy
+snowflake-sqlalchemy==1.2.0
```
Comment (villebro) [inline requirements.txt]: `extra_requires` should not be included in `requirements.txt`. This change is also pulling in libraries that are not apache compatible, at least `requests` but perhaps others, too.

**Review by villebro — CHANGES_REQUESTED** [review]

**Comment by craig-rueda** [comment]
Closing for now - this is pulling in too many external libs, which should be added into extras

**Pull request closed without merging**